### PR TITLE
[@osd/cross-platform] Adds cross-platform helpers

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <amoo_miki@yahoo.com>

### Description
* Adds the ability to get the expanded `pwd` to replace `process.cwd()`
* Moved the `standardize` function
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-config-schema/src/errors/schema_error.test.ts [hunk 1]
```diff
@@ -46,9 +48,7 @@ export const cleanStack = (stack: string) =>
       }
 
       const path = parts[1];
-      // Cannot use `standardize` from `@osd/utils
-      let relativePath = relative(process.cwd(), path);
-      if (process.platform === 'win32') relativePath = relativePath.replace(/\\/g, '/');
+      const relativePath = standardize(relative(PROCESS_WORKING_DIR, path));
```
Comment (joshuarrrr) [inline packages/osd-config-schema/src/errors/schema_error.test.ts:51]: nice!

File: packages/osd-cross-platform/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# `@osd/cross-platform` — OpenSearch Dashboards cross-platform helpers
+
+This package contains the helper functions to work around the differences of platforms
```
Comment (joshuarrrr) [inline packages/osd-cross-platform/README.md:3]: I think we can help disambiguate "platforms" in this context with an example.

```suggestion
This package contains the helper functions to work around the differences of target platforms (e.g. identifying paths on Linux or Windows systems).
```

File: packages/osd-cross-platform/src/process.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+let workingDir = process.cwd();
+
+if (process.platform === 'win32') {
+  try {
+    const pathFullName = execSync('powershell "(Get-Item -LiteralPath $pwd).FullName"', {
+      cwd: workingDir,
+      encoding: 'utf8',
+    })?.trim?.();
+    if (pathFullName?.length > 2) workingDir = pathFullName;
```
Comment (joshuarrrr) [inline packages/osd-cross-platform/src/process.ts:16]: What's the significance of `length > 2` here?

**Review by joshuarrrr — APPROVED** [review]
A couple minor questions/comments, but nothing blocking.

**Review by manasvinibs — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

File: packages/osd-cross-platform/src/process.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { execSync } from 'child_process';
+
+let workingDir = process.cwd();
+
+if (process.platform === 'win32') {
+  try {
+    const pathFullName = execSync('powershell "(Get-Item -LiteralPath $pwd).FullName"', {
```
Comment (manasvinibs) [inline packages/osd-cross-platform/src/process.ts:12]: So does this take care of resolving any short name path set for directories to plugin's full name path?

Comment (peterzhuamazon) [reply packages/osd-cross-platform/src/process.ts:12]: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-item?view=powershell-7.2#-literalpath

.FullName is literally same as `realpath` on linux.


Conbine both you get the actual full path no short name of Windows I believe.

**Comment by ashwin-pc** [comment]
Overriding failed test to fix an emergency failed build. Sign off @seanneumann. Patch for failed test coming soon.

**Pull request merged**